### PR TITLE
Issue #4901: Add activation ping metrics for data review

### PR DESCRIPTION
**Pull request by jonalmeida**
The [original data-review](https://github.com/mozilla-mobile/focus-android/pull/5065#issuecomment-893957972) for activation ID was postponed from a larger data review, so I've made a new one here.

### CODE REVIEW

**Comment by jonalmeida** [comment]
# Request for data collection review form

**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

1) **What questions will you answer with this data?**

- Similar to [Fenix](https://github.com/mozilla-mobile/fenix/pull/1707#issuecomment-486972209), we plan to count the exact number of activation, per distribution, per manufacturer.
- We also send our legacy telemetry ID as part of the deletion request for users who wish to have their reported Telemetry data deleted.

2) **Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?**

- The activation data will also help us evaluate the difference in activations from the other data sources which are currently being used  for activation counts (e.g. service-telemetry)

3) **What alternative methods did you consider to answer these questions? Why were they not sufficient?**

- Citing the learnings from [Fenix](https://github.com/mozilla-mobile/fenix/pull/1707#issuecomment-486972209) again, no current alternative methods. 

4) **Can current instrumentation answer these questions?**

- Currently, we have telemetry from a deprecated and outdated system which we can no longer rely on.

5) **List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox [data c](https://wiki.mozilla.org/Firefox/Data_Collection)[ollection ](https://wiki.mozilla.org/Firefox/Data_Collection)[categories](https://wiki.mozilla.org/Firefox/Data_Collection) found on the Mozilla wiki.**

| Measurement Description  | Data Collection Category | Tracking Bug # |
| ------------------------ | ------------------------ | -------------- |
| activation.activation_id | category 4               | #4901          |
| legacy_ids.client_id     | category 4               | #4901          |

6) **How long will this data be collected?**

- Always.

7) **What populations will you measure?**

-  All release, beta, and nightly users with telemetry enabled.

8) **Please provide a general description of how you will analyze this data.**

- Glean

9) **Where do you intend to share the results of your analysis?**

-  Only on Glean, mobile teams and BD have internal access.

**Comment by travis79** [comment]
I'm still waiting on the response from Trust/Legal on this, will review as soon as I can.

**Comment by travis79** [comment]
@jonalmeida sorry for taking so long... we are trying to make sure everything is right on this due to the identifiers being collected. In order to reduce confusion and clarify that the legacy-id and the activation-id will never be sent together, could you please move the legacy id to a separate review? 🙏 

Thanks so much!
Travis

**Comment by jonalmeida** [comment]
Updated the data review above to reflect only the activation ping.

**Comment by nshadowen314** [comment]
Hi @jonalmeida thanks for this request. Can you please address the following so I can complete the trust review? Thanks!
1. Explicitly state on data request that the ping with activation id will not contain client_id
2. Retention set to 6 months initially with potential to renew given BD still needs
3. Please answer data request q #9: "If this data collection is default on, what is the opt-out mechanism for users?"

**Comment by mcarare** [comment]
# Request for data collection review form

**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

1) What questions will you answer with this data?

Similar to Fenix, we plan to count the exact number of activation, per distribution, per manufacturer.
The ping with activation id does not contain client_id.

2) Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements? Some example responses:

The activation data will also help us evaluate the difference in activations from the other data sources which are currently being used for activation counts (e.g. service-telemetry).

3) What alternative methods did you consider to answer these questions? Why were they not sufficient?

Alternative methods were considered but none proved to be accurate. See https://bugzilla.mozilla.org/show_bug.cgi?id=1481215.  

4) Can current instrumentation answer these questions?

No, it cannot.

5) List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Data_Collection) found on the Mozilla wiki.   

**Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.**

<table>
  <tr>
    <td>Measurement Description</td>
    <td>Data Collection Category</td>
    <td>Tracking Bug #</td>
  </tr>
  <tr>
    <td>activation.activation_id </td>
    <td>Category 4 “Highly sensitive or clearly identifiable personal data” </td>
    <td>#4901</td>
  </tr>
</table>

6) Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.

This collection is documented in the Glean Dictionary at https://dictionary.telemetry.mozilla.org/apps/focus_android/metrics/activation_activation_id 
More info on the usage of this ID is documented at https://dictionary.telemetry.mozilla.org/apps/focus_android/pings/activation

7) How long will this data be collected?  Choose one of the following:

Permanently.

This is data that is permanently needed in order tocount activations.

8) What populations will you measure?

All release, beta, and nightly users with telemetry enabled.

9) If this data collection is default on, what is the opt-out mechanism for users?

Users can opt-out of data collection by disabling Usage and technical data from Settings -> Privacy and security -> Data choices.

10) Please provide a general description of how you will analyze this data.

This data will be analyzed by Glean internal systems.

11) Where do you intend to share the results of your analysis?

Only on Glean, mobile teams and BD have internal access.

12) Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection? If so:

There is no third-party tool proposed.

**Comment by mcarare** [comment]
@nshadowen314 I am submitting the above data request for analysis and approval. Thank you!

**Comment by mcarare** [comment]
> @mcarare I have escalated this to the proper folks and will try and expedite this as quickly as I can.

Thank you!

**Comment by travis79** [comment]
> # Request for data collection review form
> 
> **All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**
> 
>     1. What questions will you answer with this data?
> 
> 
> Similar to Fenix, we plan to count the exact number of activation, per distribution, per manufacturer. The ping with activation id does not contain client_id.
> 
>     2. Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements? Some example responses:
> 
> 
> The activation data will also help us evaluate the difference in activations from the other data sources which are currently being used for activation counts (e.g. service-telemetry).
> 
>     3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
> 
> 
> Alternative methods were considered but none proved to be accurate. See [bugzilla.mozilla.org/show_bug.cgi?id=1481215](https://bugzilla.mozilla.org/show_bug.cgi?id=1481215).
> 
>     4. Can current instrumentation answer these questions?
> 
> 
> No, it cannot.
> 
>     5. List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Data_Collection) found on the Mozilla wiki.
> 
> 
> **Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.**
> Measurement Description 	Data Collection Category 	Tracking Bug #
> activation.activation_id 	Category 4 “Highly sensitive or clearly identifiable personal data” 	#4901
> 
>     6. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
> 
> 
> This collection is documented in the Glean Dictionary at [dictionary.telemetry.mozilla.org/apps/focus_android/metrics/activation_activation_id](https://dictionary.telemetry.mozilla.org/apps/focus_android/metrics/activation_activation_id) More info on the usage of this ID is documented at [dictionary.telemetry.mozilla.org/apps/focus_android/pings/activation](https://dictionary.telemetry.mozilla.org/apps/focus_android/pings/activation)
> 
>     7. How long will this data be collected?  Choose one of the following:
> 
> 
> Permanently.
> 
> This is data that is permanently needed in order tocount activations.
> 
>     8. What populations will you measure?
> 
> 
> All release, beta, and nightly users with telemetry enabled.
> 
>     9. If this data collection is default on, what is the opt-out mechanism for users?
> 
> 
> Users can opt-out of data collection by disabling Usage and technical data from Settings -> Privacy and security -> Data choices.
> 
>     10. Please provide a general description of how you will analyze this data.
> 
> 
> This data will be analyzed by Glean internal systems.
> 
>     11. Where do you intend to share the results of your analysis?
> 
> 
> Only on Glean, mobile teams and BD have internal access.
> 
>     12. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection? If so:
> 
> 
> There is no third-party tool proposed.

# Data Review Form (to be filled by Data Stewards)

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way?

Yes, through the metrics.yaml file and the [Glean Dictionary](https://dictionary.telemetry.mozilla.org/).

2) Is there a control mechanism that allows the user to turn the data collection on and off?

Yes, through the "Send Usage Data" preference in the application settings.

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

Jon Almeida will monitor the collection.

4) Using the **[category system of data types](https://wiki.mozilla.org/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Category 4

5) Is the data collection request for default-on or default-off?

Default-on

6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?

Yes, an activation-id

7) Is the data collection covered by the existing Firefox privacy notice?

Yes, verified by M. Feldman

8) Does the data collection use a third-party collection tool? 

No

## Result

data-review+

**Review by travis79 — APPROVED** [review]
data-review+

**Comment by travis79** [comment]
One quick note, for this to serve the purpose of connecting the legacy telemetry data to the Glean data, you will either need to send the "activation-id" in legacy telemetry, or alternatively send the legacy telemetry id in the activation ping.

File: app/src/main/java/org/mozilla/focus/telemetry/ActivationPing.kt [hunk 1]
```diff
@@ -57,13 +60,13 @@ class ActivationPing(private val context: Context) {
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
     internal fun triggerPing() {
         // Generate the activation_id.
-        // Activation.activationId.generateAndSet()
+        Activation.activationId.set(UUID.fromString(TelemetryWrapper.clientId))
 
         CoroutineScope(Dispatchers.IO).launch {
             // Disabled until data-review r+
             // See: https://github.com/mozilla-mobile/focus-android/pull/5065
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/focus/telemetry/ActivationPing.kt:67]: We could remove these comments as well now.

**Review by jonalmeida — COMMENTED** [review]
I can't r+ myself, but looks good!

**Review by mcarare — APPROVED** [review]

**Review by ionutbedregeanu — APPROVED** [review]

**Pull request merged**